src/prompts-cli/index.ts

```diff
@@ -0,0 +1,281 @@
+import fs from 'fs/promises';
+
+type Templates = Record<string, string>;
+type TemplatePlaceholders = Record<string, string[]>;
+
+interface AutogenerationConfig {
+  // The name of the symbol we're autogenerating.
+  symbolName: string;
+  // The appearance of the symbol before the autogeneration script has run for the first time.
+  symbolAppearanceBeforeAutogeneration: string;
+  // The files where the symbol lives and where we'll insert the autogenerated content.
+  filesToModify: string[];
+  // The function that generates the content to insert.
+  generate: (args: {
+    templates: Templates;
+    templatePlaceholders: TemplatePlaceholders;
+  }) => string;
+}
+
+// The directory this package will be in once someone installs it.
+// We use this to make sure we ignore our own files when searching
+// for the templates directory.
+const THIS_PACKAGE_DIRECTORY_NAME = 'node_modules/@autoblocks/client';
+
+// We autogenerate these based on the templates directory.
+// See src/prompts/autogenerated.ts.
+const AUTOGENERATED_INTERFACE_NAME = '__AutogeneratedTemplatePathToParameters';
+const AUTOGENERATED_VARIABLE_NAME = '__autogeneratedTemplatePathToTemplate';
+
+const autogenerationConfigs: AutogenerationConfig[] = [
+  {
+    symbolName: AUTOGENERATED_INTERFACE_NAME,
+    symbolAppearanceBeforeAutogeneration: `interface ${AUTOGENERATED_INTERFACE_NAME} {\n}`,
+    filesToModify: ['../prompts/index.d.ts', '../prompts/index.d.mts'],
```

Comment:
```suggestion
    filesToModify: ['../prompts/index.d.ts', '../prompts/index.d.mjs'],
```
?

Reply:
or .mjs doesn't seem right either haven't seen mts before

Reply:
or actually is it just `index.d.ts` here?

Reply:
No we need to do both type files here (this is the interface, not the value)

Reply:
Oh i see, i didn't recognize the .mts extension. Just looked at the dist folder though and see it.